pkg/controller/workspaces/workspaces.go

```diff
@@ -148,41 +167,127 @@ func (w *WorkspacesController) StartWorkspace(token string, cheURL string, works
+func (w *WorkspacesController) StopWorkspace(keycloakUrl string, cheURL string, workspaceID string) (err error) {
+	request, err := http.NewRequest("DELETE", cheURL+"/api/workspace/"+workspaceID+"/runtime", nil)
 
 	if err != nil {
 		w.Logger.Error("Failed to stop workspace", zap.Error(err))
 	}
 
+	var token string
+	if token, err = w.KeycloakToken(keycloakUrl); err != nil {
+		w.Logger.Error("Failed to get user token ", zap.Error(err))
+	}
+
 	request.Header.Add("Authorization", "Bearer "+token)
 	request.Header.Add("Content-Type", "application/json")
 
-	_ , err = w.httpClient.Do(request)
-	time.Sleep(1 * time.Minute)
+	_, err = w.httpClient.Do(request)
+
 	return err
 }
 
 // DeleteWorkspace delete a workspace from a given workspace_id
-func (w *WorkspacesController) DeleteWorkspace(token string, cheURL string, workspaceID string) (err error) {
-	request, err := http.NewRequest("DELETE", cheURL + "/api/workspace/" + workspaceID, nil)
+func (w *WorkspacesController) DeleteWorkspace(keycloakUrl string, cheURL string, workspaceID string) (err error) {
+	request, err := http.NewRequest("DELETE", cheURL+"/api/workspace/"+workspaceID, nil)
 
 	if err != nil {
 		w.Logger.Error("Failed to delete workspace", zap.Error(err))
 	}
 
+	var token string
+	if token, err = w.KeycloakToken(keycloakUrl); err != nil {
+		w.Logger.Error("Failed to get user token ", zap.Error(err))
+	}
+
 	request.Header.Add("Authorization", "Bearer "+token)
 	request.Header.Add("Content-Type", "application/json")
 
-	_ , err = w.httpClient.Do(request)
+	_, err = w.httpClient.Do(request)
+
+	return err
+}
+
+// creating get requested with always fresh user token
+func (w *WorkspacesController) getRequest(keycloakUrl string, cheUrl string, workspaceID string) *http.Request {
+	request, err := http.NewRequest("GET", cheUrl+"/api/workspace/"+workspaceID, nil)
+
+	if err != nil {
+		w.Logger.Error("Failed to create request for obtaining workspace status ", zap.Error(err))
+	}
+
+	var token string
+	if token, err = w.KeycloakToken(keycloakUrl); err != nil {
+		w.Logger.Error("Failed to get user token ", zap.Error(err))
+	}
 
-	time.Sleep(1 * time.Minute)
+	request.Header.Add("Authorization", "Bearer "+token)
+	return request
+}
+
+// WaitWorkspaceStatusViaApi waits for workspace to have desired status in specified timeout
+func (w *WorkspacesController) WaitWorkspaceStatusViaApi(keycloakUrl string, cheURL string, workspaceID string, desiredStatus string, timeoutInSeconds int) (err error) {
+	var result map[string]interface{}
+
+	startTime, endTime := time.Now(), time.Now()
+
+	request := w.getRequest(keycloakUrl, cheURL, workspaceID)
+
+	status := "undefined"
+	response, err := w.httpClient.Do(request)
+
+	if err != nil {
+		w.Logger.Error("Failed to obtain workspace status ", zap.Error(err))
+	}
+
+	err = json.NewDecoder(response.Body).Decode(&result)
+
+	timeouted := false
+	fmt.Println("Waiting for workspace to be " + desiredStatus + ". Tick is set to 5 sec.")
```

Comment:
Would be better to use w.Logger.Info(...)

Reply:
Changed.